sdk/servicebus/Microsoft.Azure.ServiceBus/tests/Performance/Microsoft.Azure.RabbitMQ.Client.Performance/Program.cs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.RabbitMQ.Client.Performance
+{
+    using global::RabbitMQ.Client;
+    using global::RabbitMQ.Client.Events;
+    using System;
+    using System.Diagnostics;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    class Program
+    {
+        private static readonly Stopwatch _stopwatch = Stopwatch.StartNew();
+
+        private static readonly byte[] _payload = new byte[1024];
+        private const string _consumerTag = "testConsumerTag";
+
+        private static long _messages;
+        private static long _messagesReceived;
+
+        static async Task Main(string[] args)
+        {
+            var messages = (args.Length >= 1 ? long.Parse(args[0]) : 10);
+
+            var receive = (args.Length >= 2 && String.Equals(args[1], "receive", StringComparison.OrdinalIgnoreCase));
+
+            var connectionString = Environment.GetEnvironmentVariable("SERVICE_BUS_CONNECTION_STRING");
+            var entityPath = Environment.GetEnvironmentVariable("SERVICE_BUS_QUEUE_NAME");
+
+            var writeResultsTask = WriteResults(messages);
+
+            RunTest(connectionString, entityPath, messages, receive);
+
+            await writeResultsTask;
+        }
+
+        private static void RunTest(string connectionString, string entityPath, long messages, bool receive)
+        {
+            var hostname = Regex.Match(connectionString, "//([^/]*)/").Groups[1].Value;
+
+            var factory = new ConnectionFactory() { HostName = hostname, Port = 5671 };
+            factory.Ssl.Enabled = true;
+
+            if (!String.IsNullOrEmpty(Environment.GetEnvironmentVariable("SERVICE_BUS_ALLOW_UNAUTHORIZED")))
+            {
+                // Allow self-signed certs
+                factory.Ssl.CertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true;
+            }
+
+            using (var connection = factory.CreateConnection())
+            using (var channel = connection.CreateModel())
+            {
+                channel.QueueDeclare(queue: entityPath, durable: false, exclusive: false, autoDelete: false, arguments: null);
+
+                if (receive)
+                {
+                    ExecuteReceives(channel, entityPath, messages);
+                }
+                else
+                {
+                    ExecuteSends(channel, entityPath, messages);
+                }
+            }
+        }
+
+        private static void ExecuteSends(IModel channel, string entityPath, long messages)
```

Comment:
Looks good.
@mikeharder  Is there a way to know the number of accepted and rejected messages like we do in the rhea-promise test ?

Reply:
https://github.com/mikeharder/azure-sdk-for-js/blob/9d21c89adff6dacff2a520283f6f196e37fbb716/sdk/servicebus/service-bus/test/perf/rhea-promise/send.ts#L102